src/Extensions/MemberExtension.php

```diff
@@ -1,61 +1,57 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Firesphere\GraphQLJWT\Extensions;
 
+use Firesphere\GraphQLJWT\Model\JWTRecord;
 use SilverStripe\Core\Convert;
-use SilverStripe\Forms\CheckboxField;
 use SilverStripe\Forms\FieldList;
 use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\HasManyList;
 use SilverStripe\Security\Member;
 use stdClass;
 
 /**
  * Class MemberExtension
  * Add a unique token to the Member for extra validation
+ *
+ * @property Member|MemberExtension $owner
+ * @method HasManyList|JWTRecord[] AuthTokens()
  */
 class MemberExtension extends DataExtension
 {
-    private static $db = [
-        'JWTUniqueID' => 'Varchar(255)',
-    ];
+    /**
+     * List of names of extra subject fields to add to JWT token
+     *
+     * @config
+     * @var array
+     */
+    private static $jwt_subject_fields = [];
 
-    private static $indexes = [
-        'JWTUniqueID' => 'unique'
+    /**
+     * @config
+     * @var array
+     */
+    private static $has_many = [
+        'AuthTokens' => JWTRecord::class,
     ];
 
     public function updateCMSFields(FieldList $fields)
     {
-        parent::updateCMSFields($fields);
-        $fields->removeByName(['JWTUniqueID']);
-        if ($this->owner->JWTUniqueID) {
-            $fields->addFieldsToTab(
-                'Root.Main',
-                [
-                    CheckboxField::create('reset', 'Reset the Token ID to disable this user\'s remote login')
-                ]
-            );
-        }
-    }
-
-    public function onBeforeWrite()
-    {
-        parent::onBeforeWrite();
-        if ($this->owner->reset) {
-            $this->owner->JWTUniqueID = null;
-        }
+        $fields->removeByName('AuthTokens');
     }
 
     /**
      * Option to add data to the JWT Subject
      *
      * @return string
      */
-    public function getJWTData()
+    public function getJWTData(): string
     {
         $data = new stdClass();
         $identifier = Member::config()->get('unique_identifier_field');
         $extraFields = Member::config()->get('jwt_subject_fields');
 
+        $data->type = 'member';
```

Comment:
Could you make this `ClassLoader::shortClassName($this->owner->ClassName);`? I think it's a tad cleaner, allowing for extended members to have their own type?